src/pkg/util/fs/proc/proc.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package proc
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// HasFilesystem returns whether kernel support filesystem or not
+func HasFilesystem(fs string) (bool, error) {
+	p, err := os.Open("/proc/filesystems")
+	if err != nil {
+		return false, fmt.Errorf("can't open /proc/filesystems: %s", err)
+	}
+	defer p.Close()
+
+	suffix := "\t" + fs
+	scanner := bufio.NewScanner(p)
+	for scanner.Scan() {
+		if strings.HasSuffix(scanner.Text(), suffix) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// ParseMountInfo parses mountinfo pointing to path and returns a map
+// of parent mount points with associated child mount points
+func ParseMountInfo(path string) (map[string][]string, error) {
+	mp := make(map[string][]string)
+	mountlist := make(map[string][]string)
+
+	p, err := os.Open(path)
+	if err != nil {
+		return mp, fmt.Errorf("can't open %s: %s", path, err)
+	}
+	defer p.Close()
+
+	scanner := bufio.NewScanner(p)
+	for scanner.Scan() {
+		splitted := strings.Split(scanner.Text(), " ")
+		mountlist[splitted[0]] = splitted
+	}
+	for k := range mountlist {
```

Comment:
Not sure I understand the logic going on in this `for` block. Could you clarify? Is this looking to see if the source of a mount is a mountpoint itself (which would be stored in the `mountlist` then)?

Reply:
The mountlist contains key / line of mountinfo, key corresponding to the first column which is a unique number identifying the mount point, in the second loop each keys of mountlist is searched for a match in the second column, if matched so it's a children of the mount point itself

Reply:
OK figured, this makes sense!